{{cookiecutter.github_repository}}/{{cookiecutter.main_module}}/graphql/middleware.py

```diff
@@ -0,0 +1,26 @@
+from django.contrib.auth import authenticate
+from django.contrib.auth.middleware import get_user
+from django.contrib.auth.models import AnonymousUser
+from {{cookiecutter.main_module}}.users.auth.utils import get_http_authorization
+
+
+def _authenticate(request):
+    is_anonymous = not hasattr(request, "user") or request.user.is_anonymous
+    return is_anonymous  and get_http_authorization(request) is not None
+
+
+class JSONWebTokenMiddleware:
```

Comment:
Is this used anywhere?

Reply:
I think it's used in Graphene middleware in the settings: https://github.com/Fueled/django-init/pull/449/files/a74c861a22ff8b2e677f192fcb81b94e7b7c7065#diff-1d15a6d7fa02d8afb97dde149aa090cf95bb2080e98bfd1f395c48da9643a071R135

Reply:
Yes, it's used in graphene settings.
```
GRAPHENE = {
    # The location of the top-level Schema class.
    "SCHEMA": "{{ cookiecutter.main_module }}.graphql.api.schema",

    # The maximum size of objects that can be requested through a relay connection.
    "RELAY_CONNECTION_MAX_LIMIT": 100,
    "MIDDLEWARE": [
        "{{ cookiecutter.main_module }}.graphql.middleware.JSONWebTokenMiddleware"
    ],
}
```